rmw_fastrtps_cpp/src/subscription.cpp

```diff
@@ -103,6 +129,351 @@ create_subscription(
 
   /////
   // Get RMW Type Support
+
+  // Short-circuit for runtime-type subscriptions
+  const rosidl_message_type_support_t * type_support = get_message_typesupport_handle(
+    type_supports, rmw_get_dynamic_typesupport_identifier());
+  if (type_support) {
+    return __create_dynamic_subscription(
+      participant_info, type_support, topic_name, qos_policies, subscription_options, keyed);
+  }
+  // In the case it fails to find, rosidl_typesupport emits an error message
+  rcutils_reset_error();
+
+  return __create_subscription(
+    participant_info, type_supports,
+    topic_name, qos_policies, subscription_options, keyed);
+}
+
+// =================================================================================================
+// CREATE RUNTIME SUBSCRIPTION
+// =================================================================================================
+rmw_subscription_t *
+__create_dynamic_subscription(
+  CustomParticipantInfo * participant_info,
+  const rosidl_message_type_support_t * type_support,
+  const char * topic_name,
+  const rmw_qos_profile_t * qos_policies,
+  const rmw_subscription_options_t * subscription_options,
+  bool keyed)
+{
+  // NOTE(methylDragon): The dynamic type deferred case is !! NOT SUPPORTED !!
+  //                     This is because it's difficult as-is to create a subscription without
+  //                     already having the type. Too much restructuring is needed elsewhere to
+  //                     support deferral...
```

Comment:
Should we remove this or is this suppose to be accompanied with a check of some sort?